recipes/tbb/oneTBB/conanfile.py

```diff
@@ -0,0 +1,119 @@
+from conan.tools.cmake import CMake, CMakeToolchain
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import re
+
+required_conan_version = ">=1.37.0"
+
+class ConanFile(ConanFile):
+    name = "tbb"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/oneapi-src/oneTBB"
+    description = (
+        "oneAPI Threading Building Blocks (oneTBB) lets you easily write parallel C++"
+        " programs that take full advantage of multicore performance, that are portable, composable"
+        " and have future-proof scalability."
+    )
+    topics = ("tbb", "threading", "parallelism", "tbbmalloc")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "tbbmalloc": [True, False],
+        "tbbproxy": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "tbbmalloc": False,
+        "tbbproxy": False,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if tools.Version(self.version) < "2021.2.0":
+            del self.options.shared
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.get_safe("shared", True):
+            del self.options.fPIC
+        if (self.settings.os == "Macos" and
+           self.settings.compiler == "apple-clang" and
+           tools.Version(self.settings.compiler.version) < "8.0"):
+            raise ConanInvalidConfiguration("%s %s couldn't be built by apple-clang < 8.0" % (self.name, self.version))
+        if not self.options.get_safe("shared", True):
+            self.output.warn("oneTBB strongly discourages usage of static linkage")
+        if self.options.tbbproxy and not (self.options.tbbmalloc and self.options.get_safe("shared", True)):
+            raise ConanInvalidConfiguration("tbbproxy needs tbbmalloc and shared options")
+
+    def package_id(self):
+        del self.info.options.tbbmalloc
+        del self.info.options.tbbproxy
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination="src")
```

Comment:
All CCI recipes extract source code in `source_subfolder`, why `src` here?

Reply:
Because it is what I've been accustom to with my company's internal packages. Generally path length is a real concern for us since Windows is our primary platform.